### ISSUE TITLE
Option to set headers for services?

### ISSUE THREAD

**Issue by nkoppisettyintu** [movio/bramble#30]
@nmaquet Is there a way to set headers for a service in the services list?

**Comment by Ackar** [comment]
@nkoppisettyintu There's currently no way to set the headers from the config file.

It is however possible to do it from a plugin. You could have a plugin similar to this:
```go
type MyPlugin struct {
	bramble.BasePlugin
}

func (p *MyPlugin) ApplyMiddlewarePublicMux(h http.Handler) http.Handler {
	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
		ctx := r.Context()
		ctx = bramble.AddOutgoingRequestsHeaderToContext(ctx, "My-Custom-Header", "some value")
		h.ServeHTTP(rw, r.WithContext(ctx))
	})
}
```
See here https://movio.github.io/bramble/#/write-plugin for more info on how to write a plugin.

I'd be interested to know what's your use case for that if you don't mind sharing it?
If it's a common pattern we might consider adding it as a built-in plugin.

**Comment by nkoppisettyintu** [comment]
Hi @Ackar, 

Bramble is trying to connect to my service to introspect the schema but my service is rejecting bramble request as it is expecting some authorization headers. This is happening even before a client request to bramble. It fails in update() method of introspection.go of bramble.

I am getting this error - "error":"the schema is missing a Query type","level":"error","msg":"unable to update service".

**Comment by Ackar** [comment]
Thanks for explaining your use case @nkoppisettyintu.

Unless you have a common authentication for all of your services, I think the plugin approach I recommended probably wouldn't work here as it would always use the same value for every service.
We would need to change the plugin interface to add a new hook on requests to services.

Just for some background, our usage of Bramble (and the way we designed it) makes it so that Bramble is handling authentication and subsequent requests to services are unauthenticated. That way we have a central point where authentication and authorization can be handled.
This is why there is currently no built-in way to authenticate with services.
